src/core/Akka.Streams/CodeGen/Dsl/GraphApply.cs

```diff
@@ -68,18 +65,18 @@ public partial class GraphDsl
 		public static IGraph<TShapeOut, TMatOut> Create<TShapeOut, TMatOut, TMat0, TMat1, TMat2, TShape0, TShape1, TShape2>(
 			IGraph<TShape0, TMat0> g0, IGraph<TShape1, TMat1> g1, IGraph<TShape2, TMat2> g2, 
 			Func<TMat0, TMat1, TMat2, TMatOut> combineMaterializers,
-			Func<GraphDsl.Builder<TMatOut>, TShape0, TShape1, TShape2, TShapeOut> buildBlock) 
+			Func<Builder<TMatOut>, TShape0, TShape1, TShape2, TShapeOut> buildBlock) 
 			where TShapeOut: Shape
 			where TShape0: Shape
 			where TShape1: Shape
 			where TShape2: Shape
 			
 		{
-			var builder = new GraphDsl.Builder<TMatOut>();
+			var builder = new Builder<TMatOut>();
 			
-			var shape0 = builder.Add(g0);
-			var shape1 = builder.Add(g1);
-			var shape2 = builder.Add(g2);
+			var shape0 = builder.Add<TShape0, TMat0, Func<TMat1, TMat2, TMatOut>>(g0, m0 => ((m1, m2) => combineMaterializers(m0, m1, m2)));
```

Comment:
This generic type explicitness looks bad from usability point of view. Any ideas how we could describe it in such way, that compiler will be able to infer types?